pkgs/development/python-modules/pyzstd/default.nix

```diff
@@ -0,0 +1,44 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  pkgs,
+  pypaInstallHook,
+  setuptoolsBuildHook,
+}:
+
+buildPythonPackage rec {
+  pname = "pyzstd";
+  version = "0.16.2";
+  # There is a pyproject.toml, but we want to dynamically link zstd, which can
+  # only be done through a setup.py argument
+  pyproject = false;
+
+  src = fetchFromGitHub {
+    owner = "Rogdham";
+    repo = "pyzstd";
+    rev = "refs/tags/${version}";
+    hash = "sha256-Az+0m1XUFxExBZK8bcjK54Zt2d5ZlAKRMZRdr7rPcss=";
+  };
+
+  nativeBuildInputs = [
+    pypaInstallHook
+    setuptoolsBuildHook
+  ];
+
+  buildInputs = [ pkgs.zstd ];
```

Comment:
```suggestion
  buildInputs = [ zstd ];
```

Reply:
For some inexplicable reason, changing that to just zstd, and adding zstd to the function arguments causes the build to fail:

```
error: builder for '/nix/store/5vk91c1mq8bjis412imigm9zaap3qach-python3.12-pyzstd-0.16.2.drv' failed with exit code 1;
       last 25 log lines:
       > running build_ext
       > +--------------------------------------------+
       > |             Pyzstd build config            |
       > +-------------------------+------------------+
       > | Pyzstd version          | 0.16.2           |
       > +-------------------------+------------------+
       > | Implementation          | C                |
       > +-------------------------+------------------+
       > | Link to zstd library    | Dynamically link |
       > +-------------------------+------------------+
       > | Enable AVX2/BMI2        | False            |
       > +-------------------------+------------------+
       > | Debug build             | False            |
       > +-------------------------+------------------+
       > | Warning as error        | False            |
       > +-------------------------+------------------+
       > building 'pyzstd.c._zstd' extension
       > creating build/temp.linux-x86_64-cpython-312/src/bin_ext
       > gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/sbc37bmgpzr7wvlbm9lgf4y6q6dyz40q-libxcrypt-4.4.36/include -fPIC -I/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/include/python3.12 -c src/bin_ext/pyzstd.c -o build/temp.linux-x86_64-cpython-312/src/bin_ext/pyzstd.o -g0 -flto
       > In file included from src/bin_ext/pyzstd.c:1:
       > src/bin_ext/pyzstd.h:12:10: fatal error: zstd.h: No such file or directory
       >    12 | #include "zstd.h"
       >       |          ^~~~~~~~
       > compilation terminated.
       > error: command '/nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/gcc' failed with exit code 1
       For full logs, run 'nix log /nix/store/5vk91c1mq8bjis412imigm9zaap3qach-python3.12-pyzstd-0.16.2.drv'.
```

Reply:
Ah, I see why now — there's also zstd in the Python package set which shadows the top-level zstd :( Leave it as `pkgs.zstd` for now